### PR TITLE
EventHub: Added support for 2021-06-01-preview API version

### PR DESCRIPTION
**Pull request by v-Ajnava**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- https://github.com/Azure/azure-rest-api-specs/pull/15233
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [X] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by v-Ajnava** [comment]
Hi @archerzz can you please review the PR and help me understanding the error here ?

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConfigurationOperations.cs [hunk 1]
```diff
@@ -85,7 +86,7 @@ internal ConfigurationOperations(EventHubManagementClient client)
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<ClusterQuotaConfigurationProperties>> PatchWithHttpMessagesAsync(string resourceGroupName, string clusterName, ClusterQuotaConfigurationProperties parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConfigurationOperations.cs:88]: This is a breaking change. Is that due to REST API change?

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConfigurationOperationsExtensions.cs [hunk 2]
```diff
@@ -35,12 +37,13 @@ public static partial class ConfigurationOperationsExtensions
             /// <param name='clusterName'>
             /// The name of the Event Hubs Cluster.
             /// </param>
-            /// <param name='parameters'>
-            /// Parameters for creating an Event Hubs Cluster resource.
+            /// <param name='settings'>
+            /// All possible Cluster settings - a collection of key/value paired settings
+            /// which apply to quotas and configurations imposed on the cluster.
             /// </param>
-            public static ClusterQuotaConfigurationProperties Patch(this IConfigurationOperations operations, string resourceGroupName, string clusterName, ClusterQuotaConfigurationProperties parameters)
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConfigurationOperationsExtensions.cs:41]: Same breaking change here.

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConsumerGroupsOperations.cs [hunk 3]
```diff
@@ -90,7 +93,7 @@ internal ConsumerGroupsOperations(EventHubManagementClient client)
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<ConsumerGroup>> CreateOrUpdateWithHttpMessagesAsync(string resourceGroupName, string namespaceName, string eventHubName, string consumerGroupName, ConsumerGroup parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/ConsumerGroupsOperations.cs:93]: Breaking change.

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/DisasterRecoveryConfigsOperations.cs [hunk 4]
```diff
@@ -83,8 +83,16 @@ internal DisasterRecoveryConfigsOperations(EventHubManagementClient client)
         /// <return>
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<IPage<AuthorizationRule>>> ListAuthorizationRulesWithHttpMessagesAsync(string resourceGroupName, string namespaceName, string alias, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<CheckNameAvailabilityResult>> CheckNameAvailabilityWithHttpMessagesAsync(string resourceGroupName, string namespaceName, string name, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/DisasterRecoveryConfigsOperations.cs:86]: Methods are changed in this class. Is that due to REST API change?

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/DisasterRecoveryConfigsOperationsExtensions.cs [hunk 5]
```diff
@@ -33,16 +33,16 @@ public static partial class DisasterRecoveryConfigsOperationsExtensions
             /// <param name='namespaceName'>
             /// The Namespace name
             /// </param>
-            /// <param name='alias'>
-            /// The Disaster Recovery configuration name
+            /// <param name='name'>
+            /// Name to check the namespace name availability
             /// </param>
-            public static IPage<AuthorizationRule> ListAuthorizationRules(this IDisasterRecoveryConfigsOperations operations, string resourceGroupName, string namespaceName, string alias)
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/DisasterRecoveryConfigsOperationsExtensions.cs:39]: Same.

File: sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/EventHubManagementClient.cs [hunk 6]
```diff
@@ -73,40 +78,45 @@ public partial class EventHubManagementClient : ServiceClient<EventHubManagement
...
+        /// </summary>
+        public virtual IConfigurationOperations Configuration { get; private set; }
+
         /// <summary>
         /// Gets the INamespacesOperations.
         /// </summary>
         public virtual INamespacesOperations Namespaces { get; private set; }
 
         /// <summary>
-        /// Gets the IConfigurationOperations.
```
Comment (archerzz) [inline sdk/eventhub/Microsoft.Azure.Management.EventHub/src/Generated/EventHubManagementClient.cs:82]: Interface change.

**Review by archerzz — CHANGES_REQUESTED** [review]
There are quite many breaking changes. @v-Ajnava Could you please confirm that it's due to REST API changes (I guess so)?

Besides, you need to update https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.Management.EventHub/AzSdk.RP.props

**Review by archerzz — APPROVED** [review]
Had a meeting with @v-Ajnava The changes are expected. They are due to changes in REST API spec.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**